src/main/java/games/strategy/engine/pbem/CredentialManager.java

```diff
@@ -0,0 +1,366 @@
+package games.strategy.engine.pbem;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.prefs.Preferences;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Provides a facility to protect a credential before writing it to storage and to subsequently unprotect a protected
+ * credential after reading it from storage.
+ *
+ * <p>
+ * The credential manager uses a user-specific master password to protect individual credentials. The master password is
+ * stored unencrypted in the user's preference tree. Thus, the master password is in an area of the user's system that
+ * has the required permissions to ensure access only by the user. The master password will automatically be created if
+ * it does not exist.
+ * </p>
+ *
+ * <p>
+ * Instances of this class are not thread safe.
+ * </p>
+ */
+final class CredentialManager implements AutoCloseable {
+  private static final String CIPHER_ALGORITHM = "AES";
+  private static final Charset MASTER_PASSWORD_CHARSET = StandardCharsets.ISO_8859_1;
```

Comment:
Good question.  The master password only consists of Base64 characters (`[a-zA-Z0-9+/=]`).  Obviously, in this case, it would be equivalent to use either UTF-8 or ISO-8859-1, but I chose the latter because that's what the Oracle `Base64` encoder/decoder classes use (see, for example, the Javadocs for [`Base64$Encoder#encodeToString()`](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Encoder.html#encodeToString-byte:A-).